ps3.py

```diff
@@ -0,0 +1,374 @@
+    """
+    Allow the user to replace all copies of one letter in the hand (chosen by user)
+    with a new letter chosen from the VOWELS and CONSONANTS at random. The new letter
+    should be different from user's choice, and should not be any of the letters
+    already in the hand.
+
+    If user provide a letter not in the hand, the hand should be the same.
+
+    Has no side effects: does not mutate hand.
+
+    For example:
+        substitute_hand({'h':1, 'e':1, 'l':2, 'o':1}, 'l')
+    might return:
+        {'h':1, 'e':1, 'o':1, 'x':2} -> if the new letter is 'x'
+    The new letter should not be 'h', 'e', 'l', or 'o' since those letters were
+    already in the hand.
+
+    hand: dictionary (string -> int)
+    letter: string
+    returns: dictionary (string -> int)
+    """
+    hand_modified = hand.copy()
+    letters_all = string.ascii_lowercase
+    while True:
+        randomized_letter = random.choice(letters_all)
+        if randomized_letter not in hand_modified.keys():
+            hand_modified[randomized_letter] = hand_modified.pop(letter)
+            return hand_modified
+
+
+def play_game(word_list):
+    """
+    Allow the user to play a series of hands
+
+    * Asks the user to input a total number of hands
+
+    * Accumulates the score for each hand into a total score for the
+      entire series
+
+    * For each hand, before playing, ask the user if they want to substitute
+      one letter for another. If the user inputs 'yes', prompt them for their
+      desired letter. This can only be done once during the game. Once the
+      substitue option is used, the user should not be asked if they want to
+      substitute letters in the future.
+
+    * For each hand, ask the user if they would like to replay the hand.
+      If the user inputs 'yes', they will replay the hand and keep
+      the better of the two scores for that hand.  This can only be done once
+      during the game. Once the replay option is used, the user should not
+      be asked if they want to replay future hands. Replaying the hand does
+      not count as one of the total number of hands the user initially
+      wanted to play.
+
+            * Note: if you replay a hand, you do not get the option to substitute
+                    a letter - you must play whatever hand you just had.
+
+    * Returns the total score for the series of hands
+
+    word_list: list of lowercase strings
+    """
+    hands_number = int(input("Enter total number of hands: "))
+    score_for_the_game = 0
+    hand_match = deal_hand(HAND_SIZE)
+    print("Current Hand:", end=' ')
+    display_hand(hand_match)
+    substitute_answer = input("Would you like to substitute a letter? ")
+    if substitute_answer == 'yes':
+        letter_substitute = input("Which letter would you like to replace: ")
+        hand_match = substitute_hand(hand_match, letter_substitute)
+    print("")
+    score_for_the_game += play_hand(hand_match, word_list)
+    hands_number -= 1
```

Comment:
Could you rewrite while to delete this part?

Reply:
+